testsuite/gnat2goto/tests/basic_exception/test.adb

```diff
@@ -0,0 +1,26 @@
+procedure Test is
+   Too_Small : exception;
```

Comment:
Please can we also check for exceptions raised by run time checks?  Obviously these will be regarded as dead code as we don't throw the exceptions needed (or catch them...) but it would be good to know it works.

Reply:
@martin-cs runtime checks are under control of the backend (i.e. us). We currently don't emit exceptions for those at all, we have specific `*_Check` handlers for those (mostly range_check).

It _might_ be worth rethinking those now that we're tackling (at least a bit of) exception handling though.

Reply:
Should it be part of this PR?

Reply:
My apologies; I wasn't clear.  I like the way you are handling exceptions.  The function call at throw / raise and `if (0) { ... }` around the handler is good.

What I would like is one test case which tries to catch a run-time exception, say from an array bounds or the like.  This will, as @hannes-steffenhagen-diffblue observes, result in a call to the `__CPROVER_Ada_range_check` function at the array dereference and this patch will mean that the handler for the range check will be marked as dead code.  That's fine and expected.  I would just like that this happens documented in a test case.  This doesn't require you to change the handling of exceptions in this patch.

Reply:
The added one is great; thank you.